Development

Snucker opened the pull request:
fix decay see #16 

Refactor code out of review.cs. Unfortunately resets some persisted variables.
Added a more detailed po and sc breakdown in summary view.
Multiply satellite coverage sc value by 50 to make it a bit more balanced.

Snucker committed:
po and sc now set to final po and sc after review, decay now calculated in final po and sc, decay rate changed

Snucker committed:
refactor

Snucker committed:
more refactoring

Snucker committed:
fix missing instance of factorVariables in Review

Snucker committed:
added more info to summary

Snucker committed:
Merge branch 'improve_display_info' into development

# Conflicts:
#	Source/StateFunding/StateFundingGlobal.cs

Snucker committed:
multiply satellite coverage reward by 50

Snucker committed:
fixed missing event handling

linuxgurugamer commented:
Can you please be a bit more specific?  You are changing 33 files, it's a lot to review.
Also, it generally isn't a good practice to change the project files in a PR, since it requires additional work for me to understand any changes and fix them for my own environment.

I'm not declining this, but I have a lot of things I'm working on, and having to go through this is somewhat unexpected, especially the number of files. 

Thanks for understanding

Snucker committed:
removed division by 4 for funds reward. probably was there to balance po and sc decay.

Snucker commented:
Hey linuxgurugamer,

Sorry about the csproj, I'm so used to commiting that file that I didn't even check.

The main bug fix is in InstanceData.addReview() where the new budgets Public Opinion(PO) and State Confidence(SC) is now set to the old budgets final PO and SC. This bug was why players always loose PO and SC each budgets since only the decay was affecting them.

Removed ApplyDecay() from ReviewManager.cs and moved the calculation to UpdateFinalSC() and UpdateFinalPO() in Review.cs because finalPO and finalSC is displayed in the summary and would not be accurate otherwise.

The formula for finalPO and finalSC is: finalPO = (PO + POEarnedThisBudget)(1 - convergenceRate)
convergenceRate used to be 0.2
This had what I thought was an unintended consequence that finalPO converges on 4 * POEarnedThisBudget so i changed it to 0.5 which would make  finalPO converges on 1 * POEarnedThisBudget.
It turns out the mod author had intended for finalPO to be 4 times bigger and had compensated by dividing the funding reward by 4 so I'm not sure this change is necessary anymore but it does make the formula a bit more intuitive.

The abstract class Factor.cs and its implementation classes are code refactored out of Review.cs
I'm not too sold on the class name but basically a factor is anything that affects PO and SC gain.
FactorVariables.cs is a class containing all the variables to persist related factors.

Most of the changes in the view classes are to conform to the above refactor and have no visible change.
StateFundingHubCurrentView.cs and ReviewView.cs was changed to add PO and SC gain for each factor and cut down on the details for space stations.

Also I changed the SC multiplier for satellite coverage from 2 to 100 to make it a bit more balanced.

That's about it in terms of changes.
Thanks for the feedback.

Snucker committed:
revert post build event changes

linuxgurugamer commented:
Ok.  I'll merge this into a branch to test, will make it easier for me to preserve the files I need to keep.  From what I can tell, only the .csprog and the AssemblyVersion.cs are an issue.
